charts/ColorSchemes.ts

```diff
@@ -65,21 +81,32 @@ function interpolateArray(scaleArr: string[]) {
 }
 
 export class ColorScheme {
-    static fromObject(name: string, colorSets: { [key: string]: Color[] }) {
+    static fromObject(
+        name: string,
+        colorSets: { [key: string]: Color[] },
+        singleColorScale?: boolean
+    ) {
         const colorSetsArray: Color[][] = []
         Object.keys(colorSets).forEach(
             numColors => (colorSetsArray[+numColors] = colorSets[numColors])
         )
-        return new ColorScheme(name, colorSetsArray)
+        return new ColorScheme(name, colorSetsArray, singleColorScale)
     }
 
     name: string
     colorSets: Color[][] // Different color sets depending on how many distinct colors you want
     isDistinct: boolean
+    singleColorScale?: boolean
 
-    constructor(name: string, colorSets: Color[][], isDistinct?: boolean) {
+    constructor(
+        name: string,
+        colorSets: Color[][],
+        singleColorScale?: boolean,
+        isDistinct?: boolean
+    ) {
         this.name = name
         this.colorSets = []
+        this.singleColorScale = singleColorScale
```

Comment:
Maybe you can go with `this.singleColorScale = !!singleColorScale` to get rid of `undefined`, like in the line below?

Reply:
done